config/initializers/geocoder.rb

```diff
@@ -0,0 +1,4 @@
+Geocoder.configure(
+  # IP address geocoding service (default :ipinfo_io)
+  ip_lookup: :ipapi_com,
```

Comment:
@alis-khadka are we aware of any rate limits or any scenario in which this will fail (and how do we fail gracefully)?

Reply:
@donrestarone 
- The default provider (`ipinfo.io`) provides [1000 requests in a day and 50.000 requests in a month](https://github.com/alexreisner/geocoder/blob/master/README_API_GUIDE.md#ipinfoio-ipinfo_io). And the provider we are using now (`ip-api.com`) [provides 45 request per minutes](https://github.com/alexreisner/geocoder/blob/master/README_API_GUIDE.md#ipinfoio-ipinfo_io). Well they have [not mentioned about the per day or per month limit for free usage](https://ip-api.com/#:~:text=How%20many%20requests%20can%20I,please%20see%20our%20pro%20service.). So, this provider should provide 64,800 requests in a day.

- Regarding the error handling, geocoder gem will rescue any exceptions raised from the internal API calls as a **default behavior**. I have left this behavior pristine as of now.
Below we can observe that the gem only logs the error but does not throw exception by default. In such case, the location details of the IP will be `null`. Here is a [test case for this scenario](https://github.com/restarone/violet_rails/pull/1560/files#diff-f94b528980f0527e80962c4ca705a427001d83a25e4dd299fc7df1626101cdc4R137-R178).

<img width="1504" alt="Screenshot 2023-05-26 at 12 38 06 PM" src="https://github.com/restarone/violet_rails/assets/25191509/e81d7c45-bf06-42bb-9287-1ca4528c30ec">